### PR TITLE
feat: `PerronIntegralPosAux`

### PR DESCRIPTION
**Pull request by llllvvuu**

### CODE REVIEW

File: PrimeNumberTheoremAnd/MellinCalculus.lean [hunk 1]
```diff
@@ -117,6 +117,32 @@ Almost by definition.
...
+
+-- TODO: upstream to mathlib ImproperIntegral.lean
+open MeasureTheory in
+lemma integral_volume_one_div_one_add_sq : ∫ (x : ℝ), 1 / (1 + x ^ 2) = π := by
+  have h_limit : Tendsto (fun i => ∫ (x : ℝ) in -i..i, 1 / (1 + x ^ 2)) atTop (𝓝 π) := by
+    simp_rw [integral_one_div_one_add_sq, arctan_neg, sub_neg_eq_add]
+    convert Tendsto.add arctan_atTop arctan_atTop
+    exact (add_halves π).symm
+  have h_normeq (x : ℝ) : ‖1 / (1 + x ^ 2)‖ = 1 / (1 + x ^ 2) := norm_of_nonneg <|
+    one_div_nonneg.mpr <| LE.le.trans (by norm_num) <| le_add_of_nonneg_right <| sq_nonneg x
```
Comment (Ruben-VandeVelde) [inline PrimeNumberTheoremAnd/MellinCalculus.lean]:
```suggestion
    one_div_nonneg.mpr <| le_trans (by norm_num) <| le_add_of_nonneg_right <| sq_nonneg x
```

Though check if `by positivity` works

File: PrimeNumberTheoremAnd/MellinCalculus.lean [hunk 2]
```diff
@@ -125,8 +151,48 @@ $$\int_\R\frac{1}{|(1+t^2)(2+t^2)|^{1/2}}dt$$
...
+  have h_twoplus_pos (t : ℝ) : 0 < 2 + t^2 := by nlinarith
+  have h_denom_pos (t : ℝ) : sqrt (1 + t ^ 2) * sqrt (2 + t ^ 2) > 0 :=
+    mul_pos (sqrt_pos_of_pos <| h_oneplus_pos t) (sqrt_pos_of_pos <| h_twoplus_pos t)
+  have h_mono1 (t : ℝ): 1 / (1 + t^2) / 2 ≤ 1 / |sqrt (1 + t^2) * sqrt (2 + t^2)| := by
+    simp_rw [div_div]
+    apply div_le_div zero_le_one rfl.le (abs_pos_of_pos <| h_denom_pos t)
+    rewrite [abs_of_pos (h_denom_pos t)]
+    sorry
+  have h_mono2 (t : ℝ) : 1 / |sqrt (1 + t^2) * sqrt (2 + t^2)| ≤ 1 / (1 + t^2) := by
+    apply div_le_div zero_le_one rfl.le (h_oneplus_pos t)
```
Comment (Ruben-VandeVelde) [inline PrimeNumberTheoremAnd/MellinCalculus.lean]:
```suggestion
    apply div_le_div zero_le_one le_rfl (h_oneplus_pos t)
```

File: PrimeNumberTheoremAnd/MellinCalculus.lean [hunk 3]
```diff
@@ -117,6 +117,49 @@ Almost by definition.
...
+private lemma intervalIntegral_one_div_one_add_sq_tendsto :
+    Tendsto (fun i => ∫ (x : ℝ) in -i..i, 1 / (1 + x ^ 2)) atTop (𝓝 π) := by
+  convert Tendsto.add arctan_atTop arctan_atTop <;> simp
+
+open MeasureTheory in
+lemma integrable_one_div_one_add_sq : Integrable fun (x : ℝ) ↦ 1 / (1 + x ^ 2) := by
+  have (x : ℝ) : ‖1 / (1 + x ^ 2)‖ = 1 / (1 + x ^ 2) := norm_of_nonneg (by positivity)
+  refine integrable_of_intervalIntegral_norm_tendsto π (fun i ↦ ?_) tendsto_neg_atTop_atBot
+    tendsto_id (by simpa only [this] using intervalIntegral_one_div_one_add_sq_tendsto)
+  cases eq_zero_or_neZero i with
```
Comment (Ruben-VandeVelde) [inline PrimeNumberTheoremAnd/MellinCalculus.lean]: Does the `NeZero` actually help or can this be `by_cases hi : i = 0`?

Comment (llllvvuu) [reply PrimeNumberTheoremAnd/MellinCalculus.lean]: that works better, thanks!

**Review by AlexKontorovich — APPROVED** [review]
Thanks!

**Pull request merged**